lib/DateTimeParser.php

```diff
@@ -86,80 +81,90 @@ public static function parseDate($date, DateTimeZone $tz = null)
     /**
      * Parses an iCalendar (RFC5545) formatted duration value.
      *
-     * This method will either return a DateTimeInterval object, or a string
-     * suitable for strtotime or DateTime::modify.
-     *
-     * @param string $duration
-     * @param bool   $asString
+     * This method will return a DateTimeInterval object
      *
-     * @return DateInterval|string
+     * @throws InvalidDataException
+     * @throws Exception
      */
-    public static function parseDuration($duration, $asString = false)
+    public static function parseDuration(string $duration): DateInterval
```

Comment:
In order for the method to return a single type, we split it in two (see `parseDurationAsString` below, which is used only in tests).